---
title: "scVAT Tutorial"
author: "Huobin Tan"
date: "`r Sys.Date()`"
output: rmarkdown::html_vignette
vignette: >
  %\VignetteIndexEntry{Vignette Title}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
---

```{r setup, include = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>"
)
```

scVAT is a Visualization Analysis Toolkit for single-cell RNA sequence data. scVAT implements a base pipeline for scRNA, including loading, preprocessing, filtering, PCA and tSNE etc. Besides, it provides:
- A Web GUI (Graphical User Interface) for visualizing all kinds of analysis results
- Graphical clustering manually based on analysis data
- Differentiating cells

## Install
You can install scVAT from GitHub with:
```{r}
#install.package("devtools")
devtools::install_github("HuobinTan/scVAT")
```

## Loading gene-cell expression matrix
There are three different functions for loading gene-cell expression matrix
- Loading .csv file
- Loading 10X H5 file
- Loading 10X data path
```{r}
# loading matrix form csv file
# gene.index: gene name's index
# cell.index: cell name's index
# gene.in.row: gene stores in a row
csv.data <- loadCSVData ("./matrix.csv", gene.index = 1, cell.index = 1, gene.in.row=TRUE)

# loading matrix from 10x H5 file
# genome: genome in the H5, such as "mm10"", "GRch38", default NULL, the first genome 
h5.data <-load10XH5("./filtered_gene_bc_matrices_h5.h5", genome = NULL)

#loading matrix from 10x outs path, including barcodes.tsv, genes.tsv and matrix.mtx
path10x.data <- load10XPath("./filtered_gene_bc_matrices/mm10")
```

## Initialize VAT Entity

```{r}
vat <- initVATEntity(data.raw = csv.data, title = "VAT", min.genes = 0, min.cells = 0,
                          cell.scale = TRUE, scale.factor = 0,
                          log.trans = TRUE, pseudocount = 1,
                          verbose = TRUE)
```

                          
PHATE is a tool for visualizing high dimensional single-cell data with natural progressions or trajectories. PHATE uses a novel conceptual framework for learning and visualizing the manifold inherent to biological systems in which smooth transitions mark the progressions of cells from one state to another. To see how PHATE can be applied to single-cell RNA-seq datasets from hematopoietic stem cells, human embryonic stem cells, and bone marrow samples, check out our preprint on BioRxiv.
Vignettes are long form documentation commonly included in packages. Because they are part of the distribution of the package, they need to be as compact as possible. The `html_vignette` output type provides a custom style sheet (and tweaks some options) to ensure that the resulting html is as small as possible. The `html_vignette` format:

- Never uses retina figures
- Has a smaller default figure size
- Uses a custom CSS stylesheet instead of the default Twitter Bootstrap style

## Vignette Info

Note the various macros within the `vignette` section of the metadata block above. These are required in order to instruct R how to build the vignette. Note that you should change the `title` field and the `\VignetteIndexEntry` to match the title of your vignette.

## Styles

The `html_vignette` template includes a basic CSS theme. To override this theme you can specify your own CSS in the document metadata as follows:

    output: 
      rmarkdown::html_vignette:
        css: mystyles.css

## Figures

The figure sizes have been customised so that you can easily put two images side-by-side. 

```{r, fig.show='hold'}
plot(1:10)
plot(10:1)
```

You can enable figure captions by `fig_caption: yes` in YAML:

    output:
      rmarkdown::html_vignette:
        fig_caption: yes

Then you can use the chunk option `fig.cap = "Your figure caption."` in **knitr**.

## More Examples

You can write math expressions, e.g. $Y = X\beta + \epsilon$, footnotes^[A footnote here.], and tables, e.g. using `knitr::kable()`.

```{r, echo=FALSE, results='asis'}
knitr::kable(head(mtcars, 10))
```

Also a quote using `>`:

> "He who gives up [code] safety for [code] speed deserves neither."
([via](https://twitter.com/hadleywickham/status/504368538874703872))
